frontend/src/metabase/nav/containers/Navbar.jsx

```diff
@@ -110,29 +110,30 @@ class SearchBar extends React.Component {
   }
 
   render() {
+    const { active, searchText } = this.state;
     return (
       <OnClickOutsideWrapper
         handleDismissal={() => this.setState({ active: false })}
       >
         <SearchWrapper
           onClick={() => this.setState({ active: true })}
-          active={this.state.active}
+          active={active}
         >
           <Icon name="search" ml={2} />
           <SearchInput
             w={1}
             py={2}
             pr={2}
             pl={1}
-            value={this.state.searchText}
+            value={searchText}
             placeholder="Search…"
             onClick={() => this.setState({ active: true })}
             onChange={e => this.setState({ searchText: e.target.value })}
             onKeyPress={e => {
-              if (e.key === "Enter") {
+              if (e.key === "Enter" && searchText.length > 0) {
```

Comment:
Typically I'd just do `searchText` instead of `searchText.length > 0` in case `searchText` somehow gets nulled.

Reply:
what if someone types a space 
shouldn't we make sure it's non-blank?